### PR TITLE
feat: Add LM Studio Model and Embeddings Component

### PR DESCRIPTION
**Pull request by EDLLT**
Fixes https://github.com/langflow-ai/langflow/issues/3973

### CODE REVIEW

File: src/backend/base/langflow/components/models/LMStudioModel.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            model_kwargs=model_kwargs,
+            model=model_name,
+            base_url=base_url,
+            api_key=api_key,
+            temperature=temperature if temperature is not None else 0.1,
+            seed=seed,
+        )
+        if json_mode:
+            if output_schema_dict:
+                output = output.with_structured_output(schema=output_schema_dict, method="json_mode")  # type: ignore
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/components/models/LMStudioModel.py:140]: I'm using it today (loving it, btw) and it seems it has to be json_schema. We need to update this (and the OpenAI component) to make this work better with BaseModel

**Review by ogabrielluiz — COMMENTED** [review]
Hey, @EDLLT 

Looking good

File: src/backend/base/langflow/components/models/LMStudioModel.py (hunk 1 above)
Comment (EDLLT) [reply src/backend/base/langflow/components/models/LMStudioModel.py:143]: > I'm using it today (loving it, btw) and it seems it has to be json_schema. We need to update this (and the OpenAI component) to make this work better with BaseModel

By OpenAI component, you mean ``OpenAIModel.py`` right?

If I understand correctly, all what needs to be done is to swap ``method="json_mode"`` for ``method="json_schema"`` in both ``LMStudioModel.py`` and ``OpenAIModel.py``

**Comment by EDLLT** [comment]
> CI error doesn't make sense to me
> 
> ```
> Error: src/backend/base/langflow/components/models/LMStudioModel.py:57:14: RUF005 Consider iterable unpacking instead of concatenation
> Error: Process completed with exit code 1.
> ```
> 
> Line it's complaining about image
> 
> Every other component I'd seen is structured like that

I guess I'll do what the dosu bot suggested
https://github.com/langflow-ai/langflow/issues/3973#issuecomment-2393976883

**Comment by ogabrielluiz** [comment]
Hey @EDLLT 

Something like this: https://github.com/langflow-ai/langflow/blob/54dd81a6812819771029a9a1fa19227fa1fe0ed5/src/backend/base/langflow/components/models/OpenAIModel.py#L30

**Comment by EDLLT** [comment]
> Hey @EDLLT 
> 
> Something like this: https://github.com/langflow-ai/langflow/blob/54dd81a6812819771029a9a1fa19227fa1fe0ed5/src/backend/base/langflow/components/models/OpenAIModel.py#L30

Hey, so I've fixed the concatenation issue(dosu bot's pretty useful id say)

All what remains is the json schema comment you had made. 
When turning on the json mode and trying compile the lm studio component, it gives me the same error about json schema. I tried changing json mode and json object to json schema instead but it still gave me the same error.

I'm not completely sure what json mode is supposed to do either so maybe it's not supported by lm studio and should be removed?

File: src/backend/base/langflow/components/models/LMStudioModel.py (hunk 1 above)
Comment (ogabrielluiz) [reply src/backend/base/langflow/components/models/LMStudioModel.py:136]: Should be

Comment (EDLLT) [reply src/backend/base/langflow/components/models/LMStudioModel.py]: > Should be

I tried that already and it gave me the same error

**Comment by EDLLT** [comment]
@ogabrielluiz Hey. Shall I remove the JSON mode from the lmstudio model component?

**Comment by EDLLT** [comment]
Okay. I have removed the JSON mode option completely. 

> @ogabrielluiz Hey. Shall I remove the JSON mode from the lmstudio model component?

Okay, given how it was deprecated anyway, I have removed it.

**Review by ogabrielluiz — APPROVED** [review]
Hey @EDLLT 

Thanks again for this. 

LGTM

**Pull request merged**